docs/design/vm_data_protection-design.md

```diff
@@ -0,0 +1,502 @@
+
+To address this limitation, we propose a Kubernetes-native, client-server solution that operates from within the guest OS. This design enables secure, authenticated, and user-driven file-level backup and restore, without exposing sensitive infrastructure details or requiring cluster-level privileges. By complementing OADP’s snapshot-based approach, this new architectural component empowers end users to manage their own data protection needs, bridging the gap between infrastructure-level and application-level protection in OpenShift environments.
+
+## Scope
+
+The scope of the **Virtual Machine Data Protection (VMDP)** feature is to introduce a secure, user-driven mechanism for file-level backup and restore operations originating from within a guest VM. This functionality is intended to complement - NOT to replace - OADP’s existing snapshot-based backup and restore capabilities for full VMs and persistent volumes. Additionally, the new mechanism extends support to file-level operations on network-mounted file systems (e.g., NFS, CIFS) accessible from the guest OS, offering users greater flexibility and control over their data protection needs.
+
+### In Scope
+
+The following components and functionalities are considered within scope for this design and its initial implementation:
+
+- **Core Functionality**: A client-server model that enables backup and restore of specific files and directories from within a guest operating system.
+- **User Interaction**: A command-line interface (CLI) available on Linux, macOS, and Windows, allowing VM users to initiate backup and restore operations.
+- **Control Plane Integration**: OpenShift-native management using Custom Resource Definitions (CRDs) and controllers to handle and track file-level backup/restore requests.
+- **Authentication and Security**: User-specific credentials are provisioned using OpenShift `Secrets` and `ConfigMaps`, and synchronized with the Kopia repository server. These credentials allow the in-guest client to authenticate directly with the Kopia server to perform backup and restore operations.
+- **Storage Integration**: Use of OADP’s existing `BackupStorageLocation` (BSL) abstraction to define and manage backup targets.
+- **Supported File Systems**: Backup and restore operations on any standard, mounted file system accessible from within the guest OS, including network file systems such as NFS and CIFS.
+- **Repository Management**: Support for multiple logical backup repositories within a single namespace, associated with a shared BSL.
+
+### Out of Scope
+
+The following are explicitly out of scope for this project phase:
+
```

Comment:
Do we have anything we can do cap or limit the space used in s3? If not let's mark it out of scope.   Customers will want this though.

Reply:
Will add to the design as out of scope: 
```
No native mechanisms for enforcing storage quotas or caps at the repository level. This functionality may be addressed in future phases through cloud provider integration or Kopia enhancements.
```